node/chainSimulator/components/testOnlyProcessingNode_test.go

```diff
@@ -81,27 +85,410 @@ func TestNewTestOnlyProcessingNode(t *testing.T) {
 		err = node.ProcessComponentsHolder.BlockProcessor().CommitBlock(header, block)
 		assert.Nil(t, err)
 	})
+	t.Run("CreateCoreComponents failure should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsTestOnlyProcessingNode(t)
+		args.Configs.GeneralConfig.Marshalizer.Type = "invalid type"
+		node, err := NewTestOnlyProcessingNode(args)
+		require.Error(t, err)
+		require.Nil(t, node)
+	})
+	t.Run("CreateCryptoComponents failure should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsTestOnlyProcessingNode(t)
+		args.Configs.GeneralConfig.PublicKeyPIDSignature.Type = "invalid type"
+		node, err := NewTestOnlyProcessingNode(args)
+		require.Error(t, err)
+		require.Nil(t, node)
+	})
+	t.Run("CreateNetworkComponents failure should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsTestOnlyProcessingNode(t)
+		args.SyncedBroadcastNetwork = nil
+		node, err := NewTestOnlyProcessingNode(args)
+		require.Error(t, err)
+		require.Nil(t, node)
+	})
+	t.Run("CreateBootstrapComponents failure should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsTestOnlyProcessingNode(t)
+		args.Configs.FlagsConfig.WorkingDir = ""
+		node, err := NewTestOnlyProcessingNode(args)
+		require.Error(t, err)
+		require.Nil(t, node)
+	})
+	t.Run("CreateStateComponents failure should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsTestOnlyProcessingNode(t)
+		args.ShardIDStr = common.MetachainShardName // coverage only
+		args.Configs.GeneralConfig.StateTriesConfig.MaxStateTrieLevelInMemory = 0
+		node, err := NewTestOnlyProcessingNode(args)
+		require.Error(t, err)
+		require.Nil(t, node)
+	})
+	t.Run("CreateProcessComponents failure should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsTestOnlyProcessingNode(t)
+		args.Configs.FlagsConfig.Version = ""
+		node, err := NewTestOnlyProcessingNode(args)
+		require.Error(t, err)
+		require.Nil(t, node)
+	})
+	t.Run("createFacade failure should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsTestOnlyProcessingNode(t)
+		args.Configs.EpochConfig.GasSchedule.GasScheduleByEpochs = nil
+		node, err := NewTestOnlyProcessingNode(args)
+		require.Error(t, err)
+		require.Nil(t, node)
+	})
 }
 
-func TestOnlyProcessingNodeSetStateShouldError(t *testing.T) {
-	args := createMockArgsTestOnlyProcessingNode(t)
-	node, err := NewTestOnlyProcessingNode(args)
-	require.Nil(t, err)
+func TestTestOnlyProcessingNode_SetKeyValueForAddress(t *testing.T) {
+	// TODO reinstate test after Wasm VM pointer fix
+	if testing.Short() {
+		t.Skip("cannot run with -race -short; requires Wasm VM fix")
+	}
+
+	t.Parallel()
```

Comment:
I would remove t.Parallel for tests involving chain simulator components that are actually doing something. They are pretty "heavy" components

Reply:
removed